pycaret/time_series/forecasting/oop.py

```diff
@@ -1782,6 +1927,12 @@ def compare_models(
             verbose=verbose,
         )
 
+        if engines is not None:
+            # Reset the models back to the default engines
+            self._set_default_model_engines(engines=initial_model_engines)
```

Comment:
This should be done in a simple context manager. Otherwise, the model engines will not be set back on exception.